scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -980,13 +980,14 @@ def __init__(self, options, default_options=None) -> None:
         for module in modules:
             soc_root = module.meta.get("build", {}).get("settings", {}).get("soc_root")
             if soc_root:
-                self.soc_roots.append(os.path.join(module.project, soc_root))
+                self.soc_roots.append(Path(module.project) / Path(soc_root))
             dts_root = module.meta.get("build", {}).get("settings", {}).get("dts_root")
             if dts_root:
-                self.dts_roots.append(os.path.join(module.project, dts_root))
+                self.dts_roots.append(Path(module.project) / Path(dts_root))
+            print(self.dts_roots)
```

Comment:
I suppose this `print()` is just for debug purposes and you might want to remove it?